com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -167,6 +167,18 @@ internal set
 
         internal readonly HashSet<ulong> Observers = new HashSet<ulong>();
 
+#if MULTIPLAYER_TOOLS
+        private string m_CachedNameForMetrics;
+#endif
+        internal string GetNameForMetrics()
```

Comment:
On the tools side, could it maintain a dictionary of NetworkObject to "metrics name".  They you could just pass the networkObject over and de-ref in the tools, and we wouldn't have to have this tools-only extra field / code.

Reply:
That's what we had originally, but the problem is the lifecycle. We have to make sure to tell the metrics to clean up the cache otherwise there will be a leaked string for every network object after it gets destroyed. On the other hand, adding a field to the network object itself keeps the memory colocated so that it will get GC'd with the network object, which seems much cleaner.